lib/pbench/test/unit/server/test_relay.py

```diff
@@ -392,3 +405,140 @@ def test_relay_tarball_connection(
+        )
+        responses.add(
+            responses.DELETE,
+            "https://relay.example.com/uri1",
+            status=status1[0]
+            if isinstance(status1[0], int)
+            else HTTPStatus.ALREADY_REPORTED,
+            body=status1[0] if isinstance(status1[0], Exception) else None,
+        )
+        responses.add(
+            responses.DELETE,
+            "https://relay.example.com/uri2",
+            status=status2[0]
+            if isinstance(status2[0], int)
+            else HTTPStatus.ALREADY_REPORTED,
+            body=status2[0] if isinstance(status2[0], Exception) else None,
+        )
+        response = client.post(
+            self.gen_uri(server_config, "https://relay.example.com/uri1"),
+            query_string={"delete": "true"},
+            headers=self.gen_headers(pbench_drb_token),
+        )
+        assert (
+            response.status_code == HTTPStatus.CREATED
+        ), f"Unexpected result, {response.text}"
+        expected_notes = [
+            "Identified benchmark workload 'unknown'.",
+            "Expected expiration date is 2025-06-30.",
+        ]
+        if status1[0] != HTTPStatus.OK:
+            expected_notes.append(
+                f"Unable to remove relay file https://relay.example.com/uri1: '{status1[1]}'"
+            )
+        if status2[0] != HTTPStatus.OK:
+            expected_notes.append(
+                f"Unable to remove relay file https://relay.example.com/uri2: '{status2[1]}'"
+            )
+        assert response.json == {
+            "message": "File successfully uploaded",
+            "name": name,
+            "resource_id": md5,
+            "notes": expected_notes,
+        }
+        assert len(responses.calls) == 4
+        assert (
+            response.headers["location"]
+            == f"https://localhost/api/v1/datasets/{md5}/inventory/"
+        )
+
+        audit = Audit.query()
+        assert len(audit) == 2
+        assert audit[0].id == 1
+        assert audit[0].root_id is None
+        assert audit[0].operation == OperationCode.CREATE
+        assert audit[0].status == AuditStatus.BEGIN
+        assert audit[0].name == "relay"
+        assert audit[0].object_type == AuditType.DATASET
+        assert audit[0].object_id == md5
+        assert audit[0].object_name == name
+        assert audit[0].user_id == DRB_USER_ID
+        assert audit[0].user_name == "drb"
+        assert audit[0].reason is None
+        assert audit[0].attributes == {
+            "access": "private",
+            "metadata": {"global.pbench.test": "data"},
+        }
+        assert audit[1].id == 2
+        assert audit[1].root_id == 1
+        assert audit[1].operation == OperationCode.CREATE
+        assert audit[1].status == AuditStatus.SUCCESS
+        assert audit[1].name == "relay"
+        assert audit[1].object_type == AuditType.DATASET
+        assert audit[1].object_id == md5
+        assert audit[1].object_name == Dataset.stem(file)
```

Comment:
Is the comparison value in line 536 supposed to be different from the one used in line 521?

Reply:
I cloned the old `test_relay` and made changes for the new cases; this is all untouched. But, yeah, presumably at some point I added the `name` local and neglected to change the analog of this check. I guess I might as well clean that up.